internal/events/logprocessor.go

```diff
@@ -28,13 +28,15 @@ import (
 )
 
 type logEntry struct {
-	Address          ethbinding.Address   `json:"address"`
-	BlockNumber      ethbinding.HexBigInt `json:"blockNumber"`
-	TransactionIndex ethbinding.HexUint   `json:"transactionIndex"`
-	TransactionHash  ethbinding.Hash      `json:"transactionHash"`
-	Data             string               `json:"data"`
-	Topics           []*ethbinding.Hash   `json:"topics"`
-	Timestamp        uint64               `json:"timestamp,omitempty"`
+	Address          ethbinding.Address     `json:"address"`
+	BlockNumber      ethbinding.HexBigInt   `json:"blockNumber"`
+	TransactionIndex ethbinding.HexUint     `json:"transactionIndex"`
+	TransactionHash  ethbinding.Hash        `json:"transactionHash"`
+	Data             string                 `json:"data"`
+	Topics           []*ethbinding.Hash     `json:"topics"`
+	Timestamp        uint64                 `json:"timestamp,omitempty"`
+	InputMethod      string                 `json:"inputMethod,omitempty"`
+	InputArgs        map[string]interface{} `json:"inputArgs,omitempty"`
```

Comment:
I couldn't entirely figure out the reason for the existence of both `eventData` and `logEntry`. The two structs seem almost identical and most data is just copied from one to the other, which seems less efficient than it could be.

Reply:
@peterbroadhurst did you have any info on this distinction? Currently I'm performing the transaction lookup in `subscription.go` while building the `logEntry`, and everything is copied from there to the `eventData`.

One other option would be to not put any of it on `logEntry`, and to only do the parsing when building `eventData` - this would require passing the `ContractResolver` and `ABILocation` from the subscription down into `logprocessor.go`.